src/pages/submit/grant.tsx

```diff
@@ -79,9 +80,30 @@ function parseStringsAsNumbers(grantRequest: GrantRequest) {
   }
 }
 
+function handleCancel() {
+  if ((window as any).routeUpdate) {
+    window.history.back()
+  } else {
+    navigate(locations.submit())
+  }
+}
+
 export default function SubmitGrant() {
   const t = useFormatMessage()
   const [account, accountState] = useAuthContext()
+  useEffect(() => {
+    const params = new URLSearchParams(location.search)
+    let category: NewGrantCategory | null = null
+    try {
```

Comment:
Why don't...?
```
    try {
      const category = toNewGrantCategory(params.get('category'))
      patchGrantRequest({ category })
    } catch (error) {
      console.error(error)
    }
```

Reply:
If `toNewGrantCategory` throws an error, the patch would not be executed and it is necessary that it is done regardless of whether it fails or not, that is why it is in finally